storage/localstore/mode_set.go

```diff
@@ -230,8 +124,198 @@ func (db *DB) set(mode chunk.ModeSet, addr chunk.Address) (err error) {
 	if err != nil {
 		return err
 	}
-	if triggerPullFeed {
-		db.triggerPullSubscriptions(db.po(item.Address))
+	for po := range triggerPullFeed {
+		db.triggerPullSubscriptions(po)
+	}
+	return nil
+}
+
+// setAccess sets the chunk access time by updating required indexes:
+//  - add to pull, insert to gc
+// Provided batch and binID map are updated.
+func (db *DB) setAccess(batch *leveldb.Batch, binIDs map[uint8]uint64, addr chunk.Address, po uint8) (gcSizeChange int64, err error) {
+
+	item := addressToItem(addr)
+
+	// need to get access timestamp here as it is not
+	// provided by the access function, and it is not
+	// a property of a chunk provided to Accessor.Put.
+	i, err := db.retrievalDataIndex.Get(item)
+	switch err {
+	case nil:
+		item.StoreTimestamp = i.StoreTimestamp
+		item.BinID = i.BinID
+	case leveldb.ErrNotFound:
+		db.pushIndex.DeleteInBatch(batch, item)
```

Comment:
can you explain a bit better the case here? if it isn't in the retrieval data index does this not infer that the index is corrupted?

Reply:
Yes, this case can happen only if the chunk is removed by gc in between it is got from localstore and its access time is set. In that case, the error would not be relevant. Maybe we could do something else in this case? Maybe just return nil?